text/0002-numeric-values.md

```diff
@@ -0,0 +1,485 @@
+
+- `+`: Addition
+- `-`: Subtraction
+- `*`: Multiplication
+- `/`: Division
+- `%`: Remainder
+- `<<`: Bitwise left shift
+- `>>`: Bitwise right shift
+- `&`: Bitwise AND
+- `|`: Bitwise OR
+- `^`: Bitwise XOR
+
+The following unary operators are available:
+
+- `-`: Negation
+- `~`: Bitwise NOT
+
+The above operators share the [same precedence as JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence), to reduce confusion.
+
+These all function as JavaScript does, except bitwise operators do not [convert the value to a 32-bit integer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Left_shift#description). Instead, it will keep the number at whatever it's bit width is.
+
+```js
+0 ^ 4294967296;
+// JavaScript -> 0
+// Blight     -> 4294967296
+```
+
+Since fixed-width integer types have a specific minimum and maximum, it is undefined behavior to error to overflow or underflow when doing arithmatic. In a safe build, this will safely crash with a stack trace and information on what values were provided, but at runtime this will cause undefined behavior.
+
+The following operations will cause undefined behavior:
+
+- `+`: If the result of the addition is above `Int.max`
+- `-`: If the result of the subtraction is above `Int.min`.
+- `*`: If the result of the multiplication is above `Int.max` or below `Int.min`.
+- `/`: If there is a remainder when dividing.
+- `/`: If the divisor is `0`.
+- `%`: If the divisor is `0`.
+- Unary `-` can underflow if when negating `Int.max`.
+
+## Floats
+
+There is one floating point type: `f64`, which is a 64-bit floating point that follows the [IEEE 754 standard](https://en.wikipedia.org/wiki/IEEE_754).
+
+## Built-in Properties on the Float Type
+
+`f64` has built in properties. These are accessed by using property access notation.
+
+- `f64.min`: Constant `1.7976931348623157e+308`
+- `f64.max`: Constant `5e-324`
+- `f64.inf`: Constant `Infinity`
+
+## Built-in Properties on Float Values
+
+Similarly to integers, floats have built-in properties and methods. The variable name `val` is a placeholder for the float value used.
+
+- `val.to(T)`: Converts the float's value to a the given type. T can be
+  - An integer type like `u8`, which it will convert the value to that type, rounding the float if needed.
+    - It is safety-checked undefined behavior if the float is out of range of `T.min`..`T.max`.
+    - It is safety-checked undefined behavior if the float is not an integer.
+  - `f64`, which it will do nothing.
+- `val.round()`: Round this float to the nearest integer, while keeping it a float.
+- `val.floor()`: Round this float down to the nearest integer, while keeping it a float.
+- `val.ceil()`: Round this float up to the nearest integer, while keeping it a float.
+- `val.pow(y)`: Raise this float to the power of `y`.
+  - It is safety-checked undefined behavior to call `1.pow(f64.inf)`.
```

Comment:
Add a safety check for 0^0

Reply:
0.pow(0) is 1. i will clarify this identity in the document.